src/pages/Tags/Tags.tsx

```diff
@@ -1,9 +1,242 @@
+      ariaLabel: t('tag.tag_name'),
+      label: 'Nimetus',
+      name: `tagInput.${uuidv4()}`,
+      placeholder: t('tag.tag_input'),
+      type: 'text',
+      rules: {
+        required: true,
+        validate: tagInputValidator,
+      },
+      className: classes.tagInputField,
+      disabled: !includes(userPrivileges, Privileges.AddTag),
+    },
+  ]
+
+  const categoryFields: FieldProps<FormValues>[] = [
+    {
+      inputType: InputTypes.Selections,
+      name: 'tagCategorySelection',
+      ariaLabel: t('tag.select_tag_category'),
+      options: updatedDataWithoutSkillsObject,
+      placeholder: t('tag.select_tag_category'),
+      multiple: true,
+      rules: {
+        required: true,
+      },
+      buttons: true,
+      disabled: !includes(userPrivileges, Privileges.AddTag),
+    },
+  ]
+
+  const [inputFields, setInputFields] = useState(tagFields)
+
+  const addInputField = () => {
+    setInputFields([
+      ...inputFields,
+      {
+        inputType: InputTypes.Text,
+        ariaLabel: t('tag.tag_name'),
+        label: 'Nimetus',
+        name: `tagInput.${uuidv4()}`,
+        placeholder: t('tag.tag_input'),
+        type: 'text',
+        rules: {
+          required: true,
+          validate: tagInputValidator,
+        },
+        className: classes.tagInputField,
+      },
+    ])
+  }
+
+  const onTagsSubmit: SubmitHandler<FormValues> = useCallback(
+    async (values) => {
+      const { tagCategorySelection, tagInput } = values
+      const transformedObject = flatMap(tagInput, (tagInputValue) =>
+        map(tagCategorySelection, (tagCategoryValue) => {
+          return {
+            type: tagCategoryValue,
+            name: tagInputValue,
+          }
+        })
+      )
+      const payload: TagsType = {
+        tags: transformedObject,
+      }
+
+      try {
+        await createTags(payload)
+        showNotification({
+          type: NotificationTypes.Success,
+          title: t('notification.announcement'),
+          content: t('success.tag_added'),
+        })
+        reset()
+      } catch (errorData) {
+        showValidationErrorMessage(errorData)
+      }
+    },
+    [createTags, reset, t]
+  )
+
+  const groupedCategoryData = groupBy(tags, 'type')
+  const uniqueCategoryTypes = Object.keys(groupedCategoryData)
+
+  const areInputFieldsDirty = !!(
+    dirtyFields.tagCategorySelection && dirtyFields.tagInput
+  )
+
+  if (isFetchingTags) {
+    return <Loader loading />
+  }
+
+  return (
+    <>
+      <div className={classes.tagsHeader}>
+        <h1>{t('tag.tag_management')}</h1>
+        <Tooltip
+          title={t('cheat_sheet.user_management.title')}
+          modalContent={<TagsCheatSheet />}
+        />
+      </div>
+      <Container className={classes.tagsContainer}>
+        <div>
+          <h4 className={classes.addingTag}>{t('tag.adding_tag')}</h4>
+          <p>{t('tag.naming_tag')}</p>
+        </div>
+
+        <div className={classes.addingTagsSeparator} />
```

Comment:
[blocker]

no need for separator div, just add` border-right: 1px solid $color-background-dark;` for naming tag container or border-left for tagsSection div